server/api/src/rsGeneralAdmin.cpp

```diff
@@ -431,18 +432,27 @@ int _addChildToResource(generalAdminInp_t* _generalAdminInp, rsComm_t* _rsComm)
     log_api::info(
         R"__(rsGeneralAdmin add child "{}" to resource "{}")__", _generalAdminInp->arg3, _generalAdminInp->arg2);
 
-    if ( ( result = chlAddChildResc( _rsComm, resc_input ) ) != 0 ) {
+    if ((result = chlAddChildResc(_rsComm, resc_input)) == 0) {
+        try {
+            log_api::info("ADDING CHILD [{}] TO RESOURCE [{}] ::: rescChildren = [{}]", _generalAdminInp->arg3, _generalAdminInp->arg2, rescChildren);
+            resc_mgr.add_child_to_resource(_generalAdminInp->arg2, _generalAdminInp->arg3);
```

Comment:
What should happen if this fails? Should everything be rolled back?

Reply:
It looks to me like it should.